src/main/java/org/littleshoot/proxy/impl/GlobalStateWrapperEvenLoop.java

```diff
@@ -0,0 +1,183 @@
+package org.littleshoot.proxy.impl;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import io.netty.util.concurrent.EventExecutor;
+import io.netty.util.concurrent.EventExecutorGroup;
+import io.netty.util.concurrent.Future;
+import io.netty.util.concurrent.ProgressivePromise;
+import io.netty.util.concurrent.Promise;
+import io.netty.util.concurrent.ScheduledFuture;
+
+public class GlobalStateWrapperEvenLoop implements EventExecutor {
```

Comment:
this is very simple wrapper without any modification to the logic of the main event loop. Any attempt to do that made the proxy unstable